packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -59,9 +59,122 @@ String getAdbPath([AndroidSdk existingSdk]) {
   }
 }
 
+class AndroidNdkSearchError {
+  AndroidNdkSearchError(this.reason);
+
+  /// The message explaining why NDK was not found.
+  final String reason;
+}
+
+class AndroidNdk {
+  AndroidNdk._(this.directory, this.compiler, this.compilerArgs);
+
+  /// The path to the NDK.
+  final String directory;
+
+  /// The path to the NDK compiler.
+  final String compiler;
+
+  /// The mandatory arguments to the NDK compiler.
+  final List<String> compilerArgs;
+
+  /// Locate NDK within the given SDK or throw [AndroidNdkSearchError].
+  static AndroidNdk locateNdk(String androidHomeDir) {
+    if (androidHomeDir == null) {
+      throw new AndroidNdkSearchError('Can not locate NDK because no SDK is found');
+    }
+
+    String findBundle(String androidHomeDir) {
+      final String ndkDirectory = fs.path.join(androidHomeDir, 'ndk-bundle');
+      if (!fs.isDirectorySync(ndkDirectory)) {
+        throw new AndroidNdkSearchError('Can not locate ndk-bundle, tried: ${ndkDirectory}');
+      }
+      return ndkDirectory;
+    }
+
+    String findCompiler(String ndkDirectory) {
+      String directory;
+      if (platform.isLinux) {
+        directory = 'linux-x86_64';
+      } else if (platform.isMacOS) {
+        directory = 'darwin-x86_64';
+      } else {
+        throw new AndroidNdkSearchError('Only Linux and Mac OS are supported');
```

Comment:
Ultra-nit: these days it's `macOS`, though they'll never convince me to stop calling it OS X ;-)

Reply:
Done.